arch/arm/src/armv8-r/arm_syscall.c

```diff
@@ -254,7 +261,20 @@ uint32_t *arm_syscall(uint32_t *regs)
 #endif
 
       case SYS_restore_context:
+        switch_context = true;
+
+        /* Update scheduler parameters */
+
+        nxsched_resume_scheduler(tcb);
+        break;
       case SYS_switch_context:
```

Comment:
could merge the code block of SYS_restore_context and SYS_switch_context.

Reply:
done